resource_list.go

```diff
@@ -0,0 +1,76 @@
+package bunny
+
+import "context"
+
+const (
+	// DefaultPaginationPage is the default value that is used for
+	// PaginationOptions.Page if it is unset.
+	DefaultPaginationPage = 1
+	// DefaultPaginationPerPage is the default value that is used for
+	// PaginationOptions.PerPage if it is unset.
+	DefaultPaginationPerPage = 1000
+)
+
+// PaginationOptions specifies optional parameters for List APIs.
+type PaginationOptions struct {
+	// Page the page to return
+	Page int32 `url:"page,omitempty"`
+	// PerPage how many entries to return per page
+	PerPage int32 `url:"per_page,omitempty"`
+}
+
+// PaginationReply represents the pagination information contained in a
+// List API endpoint response.
+//
+// Ex. Bunny.net API docs:
+// - https://docs.bunny.net/reference/pullzonepublic_index
+// - https://docs.bunny.net/reference/storagezonepublic_index
+type PaginationReply struct {
+	CurrentPage  *int32 `json:"CurrentPage"`
+	TotalItems   *int32 `json:"TotalItems"`
+	HasMoreItems *bool  `json:"HasMoreItems"`
+}
+
+func (p *PaginationOptions) ensureConstraints() {
+	if p.Page < 1 {
+		p.Page = DefaultPaginationPage
+	}
+
+	if p.PerPage < 1 {
+		p.PerPage = DefaultPaginationPerPage
+	}
+}
+
+// ResourceList is a generic function for listing resources
+func ResourceList[Resp any](
```

Comment:
I've updated it to do something similar to what you're suggesting. Take a look and let me know what you think.

Reply:
Good idea to invert the relationship. 

I also did a bit of research now and it seems not possible to only define that a type has to have certain struct field  embeds another structs.
Adding getter methods for PaginationReply + defining an interface which is used for the type constraint, should work.
The package user that probably can not access the PaginationReply fields directly and would have to call getter-method. I think that would be much worse too use.

I prefer the previous version, where `Resp` is any. 
With the new change the public API of the package also starts to use generics, which makes it more complex for the package user, only because we want to replace 2 functions calls with one internally.
I think that is not worth it.

Reply:
I changed it back to `any` and left the inverted relationship.